packages/components/data-source/src/fetch-list-data-source.ts

```diff
@@ -89,9 +89,12 @@ export class FetchListDataSource<T = any> extends ListDataSource<T> {
     if (options.placeholder) {
       this.placeholder = options.placeholder;
     }
+
+    // Initialize the items array, but do not emit an event yet
+    this.update(false);
   }
 
-  update(): void {
+  update(emitEvent = true): void {
```

Comment:
So when we are filtering and there is a pagination added, we don't move to the first page anymore?
![image](https://github.com/user-attachments/assets/3f809eb2-79c8-4848-8635-f6f213bb9e85)

Reply:
This is happening on `main` as well; i've created #2021 for this